tunnel/intra/doh/ipmap/ipmap.go

```diff
@@ -100,7 +102,10 @@ func (s *IPSet) add(ip net.IP) {
 // The hostname can be a domain name or an IP address.
 func (s *IPSet) Add(hostname string) {
 	// Don't hold the ipMap lock during blocking I/O.
-	resolved, _ := s.r.LookupIPAddr(context.TODO(), hostname)
+	resolved, err := s.r.LookupIPAddr(context.TODO(), hostname)
+	if err != nil {
+		log.Infof("Error resolving %s: %v", hostname, err)
```

Comment:
nit: Errorf?

Reply:
I changed this to `Warnf`.  (I try to reserve "Error" for bugs, whereas this error will occur whenever you are experiencing DNS poisoning of a resolver name.)